### ISSUE TITLE
Support for PHP 8.1 enums

### ISSUE THREAD

**Issue by ricardoboss** [laravel/serializable-closure#43]
Hi, good job with this fork!

I recently ran into a situation, where a class has an `enum` property, which prevents the call to `ReflectionClass->newInstanceWithoutConstructor()`, since an enum cannot be instantiated:

```
PHP Fatal error:  Uncaught Error: Cannot instantiate enum Project\AnEnum in ...\vendor\laravel\serializable-closure\src\Serializers\Native.php:261
Stack trace:
#0 ...\vendor\laravel\serializable-closure\src\Serializers\Native.php(261): ReflectionClass->newInstanceWithoutConstructor()
#1 ...\vendor\laravel\serializable-closure\src\Serializers\Native.php(282): Laravel\SerializableClosure\Serializers\Native::wrapClosures(Object(Project\AnEnum), Object(Laravel\SerializableClosure\Support\ClosureScope))
#2 ...\vendor\laravel\serializable-closure\src\Serializers\Native.php(122): Laravel\SerializableClosure\Serializers\Native::wrapClosures(Object(Project\ParentClass), Object(Laravel\SerializableClosure\Support\ClosureScope))
#3 [internal function]: Laravel\SerializableClosure\Serializers\Native->__serialize()
...
```

Any intention of supporting this use-case?

**Comment by nunomaduro** [comment]
Hi @ricardoboss, do you mind of making a pull request adding a test that fails? https://github.com/laravel/serializable-closure/blob/master/tests/SerializerPhp81Test.php

**Comment by nunomaduro** [comment]
I will re-open the issue once you add a failing test.

**Comment by ricardoboss** [comment]
Looks like it is a problem with my codebase... Was unable to reproduce it. Thanks for your time!

**Comment by ksassnowski** [comment]
I just encountered the same bug and was able to reproduce it with a test case. I have opened a PR to fix the issue (#47)
